### PR TITLE
Fix Dockerfile.e2e build

### PR DESCRIPTION
**Pull request by vdemeester**
This images is used to run integration and integration-cli tests on
anything that implements the docker api :). The image wasn't building
anywore :D

cc @chris-crone 

I'll probably update the CI so that it builds and push that images :angel: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: Dockerfile.e2e [hunk 1]
```diff
@@ -15,26 +15,29 @@ WORKDIR /go/src/github.com/docker/docker/
 
 # Generate frozen images
 COPY contrib/download-frozen-image-v2.sh contrib/download-frozen-image-v2.sh
-RUN contrib/download-frozen-image-v2.sh /output/docker-frozen-images \
+RUN ./contrib/download-frozen-image-v2.sh /output/docker-frozen-images \
```
Comment (thaJeztah) [inline Dockerfile.e2e]: Was this related to the fix? If so, curious what changed, or if there's a regression

File: Dockerfile.e2e [hunk 2]
```diff
@@ -15,26 +15,29 @@ WORKDIR /go/src/github.com/docker/docker/
...
 	debian:jessie@sha256:287a20c5f73087ab406e6b364833e3fb7b3ae63ca0eb3486555dc27ed32c6e60 \
 	hello-world:latest@sha256:be0cd392e45be79ffeffa6b05338b98ebb16c87b255f48e297ec7f98e123905c
 
-# Download Docker CLI binary
-COPY hack/dockerfile hack/dockerfile
-RUN hack/dockerfile/install.sh dockercli
+# Install dockercli
+# Please edit hack/dockerfile/install/<name>.installer to update them.
+COPY hack/dockerfile/install hack/dockerfile/install
+RUN for i in dockercli; \
```
Comment (cpuguy83) [inline Dockerfile.e2e]: Why the loop?

File: Dockerfile.e2e [hunk 3]
```diff
@@ -15,26 +15,29 @@ WORKDIR /go/src/github.com/docker/docker/
...
+		do ./hack/dockerfile/install/install.sh $i; \
+done
 
 # Set tag and add sources
-ARG DOCKER_GITCOMMIT
+ARG DOCKER_GITCOMMIT=undefined
```
Comment (thaJeztah) [inline Dockerfile.e2e]: perhaps instead:

```Dockerfile
ENV DOCKER_GITCOMMIT=${DOCKER_GITCOMMIT:-undefined}
```

File: Dockerfile.e2e [hunk 4]
```diff
@@ -15,26 +15,29 @@ WORKDIR /go/src/github.com/docker/docker/
...
 ENV DOCKER_GITCOMMIT=$DOCKER_GITCOMMIT
 ADD . .
 
 # Build DockerSuite.TestBuild* dependency
 RUN CGO_ENABLED=0 go build -buildmode=pie -o /output/httpserver github.com/docker/docker/contrib/httpserver
 
 # Build the integration tests and copy the resulting binaries to /output/tests
-RUN hack/make.sh build-integration-test-binary
+RUN ./hack/make.sh build-integration-test-binary
```
Comment (thaJeztah) [inline Dockerfile.e2e]: Same question here

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Some questions 😅

File: Dockerfile.e2e (hunk 2 above)
Comment (vdemeester) [reply Dockerfile.e2e]: Because of `copy/paste` 😅

File: Dockerfile.e2e (hunk 1 above)
Comment (vdemeester) [reply Dockerfile.e2e]: well, it's just a "personal" pref 😓

File: Dockerfile.e2e (hunk 3 above)
Comment (vdemeester) [reply Dockerfile.e2e]: Would work too

File: Dockerfile.e2e (hunk 2 above)
Comment (cpuguy83) [reply Dockerfile.e2e]: Oh I see I missed a directory when adding that install.sh to this one.

**Comment by vdemeester** [comment]
@thaJeztah @cpuguy83 updated :angel:

File: Dockerfile.e2e (hunk 1 above)
Comment (thaJeztah) [reply Dockerfile.e2e]: oh, it's fine to change, was just wondering if there was an issue there 😅

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**